packages/hop-node/src/gasboost/GasBoostTransaction.ts

```diff
@@ -301,31 +302,33 @@ class GasBoostTransaction extends EventEmitter implements providers.TransactionR
   }
 
   async getBumpedGasPrice (multiplier : number = this.gasPriceMultiplier): Promise<BigNumber> {
+    const maxGasPrice = this.parseGwei(this.maxGasPriceGwei)
     const marketGasPrice = await this.getMarketGasPrice()
     if (!this.isChainGasFeeBumpable()) {
-      return marketGasPrice
+      return BNMin(marketGasPrice, maxGasPrice)
     }
     const prevGasPrice = this.gasPrice || marketGasPrice
     const bumpedGasPrice = getBumpedGasPrice(prevGasPrice, multiplier)
     if (!this.compareMarketGasPrice) {
-      return bumpedGasPrice
+      return BNMin(bumpedGasPrice, maxGasPrice)
     }
-    return BNMax(marketGasPrice, bumpedGasPrice)
+    return BNMin(BNMax(marketGasPrice, bumpedGasPrice), maxGasPrice)
   }
 
   async getBumpedMaxPriorityFeePerGas (multiplier : number = this.gasPriceMultiplier): Promise<BigNumber> {
+    const priorityFeePerGasCap = this.parseGwei(this.priorityFeePerGasCap)
     const marketMaxPriorityFeePerGas = await this.getMarketMaxPriorityFeePerGas()
     if (!this.isChainGasFeeBumpable()) {
-      return marketMaxPriorityFeePerGas
+      return BNMin(marketMaxPriorityFeePerGas, priorityFeePerGasCap)
     }
     const prevMaxPriorityFeePerGas = this.maxPriorityFeePerGas || marketMaxPriorityFeePerGas
     const minPriorityFeePerGas = this.parseGwei(this.minPriorityFeePerGas)
     let bumpedMaxPriorityFeePerGas = getBumpedBN(prevMaxPriorityFeePerGas, multiplier)
     bumpedMaxPriorityFeePerGas = BNMax(minPriorityFeePerGas, bumpedMaxPriorityFeePerGas)
     if (!this.compareMarketGasPrice) {
-      return bumpedMaxPriorityFeePerGas
+      return BNMin(bumpedMaxPriorityFeePerGas, priorityFeePerGasCap)
     }
-    return BNMax(marketMaxPriorityFeePerGas, bumpedMaxPriorityFeePerGas)
+    return BNMin(BNMax(marketMaxPriorityFeePerGas, bumpedMaxPriorityFeePerGas), priorityFeePerGasCap)
```

Comment:
Same as above.

Do you know if the nodes impose a `maxPrioirtyGasFee` increase minimum (like the 10% `gasPrice` minimum)?

Reply:
I believe it does

https://twitter.com/TimBeiko/status/1413541934719770628?s=20